src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -720,6 +723,15 @@ export class ListDatabasesActionItem extends Disposable implements IActionViewIt
 				});
 	}
 
+	// Removes the DB pool instance name from the database name
+	private removePoolInstanceName(dbName: string): string {
+		if (dbName.includes('@')) {
```

Comment:
@ is a valid character in a database name so wouldn't this would break any of those scenarios? (and it seems like the logic in STS may also be broken in that case)

I'm not really familiar with this feature - but is this something we could strip out when we fetch the databases for MI pools specifically? And regardless it would be really great if we could figure out better ways to handle cases like this other than having generic checks in the core code.

Reply:
the databases returned do not have the "@" character regardless if it's MI pool or not, which was causing the issue in the first place. I'm not sure if there's an easy way to check what pool we're connected to.

Reply:
Charles is correct that a database can contain an '@' symbol apparently (e.g. `CREATE DATABASE TestDb@102` executed without error).  In this case, I think we can just strip-off the suffix without first determining the connect is to a DW server with DB Pools enabled.  I guess in the case that the DB contains an '@' and the user is connecting to a pool we should strip-off the final '@...'.  It may be the case that we want to strip off the pool from the DB Name in STS, but then we'd need to retain the pool details somewhere else so we can use it during reconnections.

Reply:
I was thinking of using the `pooling` boolean option in connection options to indicate this in STS. Do you know if that option is really meant for managed instance pooling or some other kind of pooling in connections? It's usually always set to false